keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -324,49 +333,101 @@ def next(prompt, state, index):
+        model, inputs should be raw strings. If `preprocessor` is `None`, inputs
+        should be a dictionary with keys `"token_ids"` and `"padding_mask"` and
+        be padded to the maximum length of generated output sequences.
 
         Args:
-            prompt: a string, string Tensor or string RaggedTensor. The prompt
-                text for generation.
+            inputs: a string `tf.Tensor`, a `tf.data.Dataset` of strings, or
+                list of python strings.
             max_length: int. The max length of generated sequence.
-            sampler: a string or `keras_nlp.samplers.Sampler` instance. The
-                sampler to be used for text generation.
+            batch_size: int. Only pass if `inputs` is a `tf.Tensor`. If set,
+                `inputs` will be split into `batch_size` batches handled
+                successively.
+            return_tensors: bool. Only used if a `preprocessor` is attached to
+                the model. If `True`, `tf.Tensor` outputs will be returned. If
+                `False`, a python list of strings will be return.
+
+        Returns:
+            A string or string list if `preprocessor` is set, and a integer
+            tensor of token IDs if `preprocessor` is `None`.
         """
-        if self.preprocessor is None:
+        input_is_scalar = False
+
+        if isinstance(inputs, tf.data.Dataset) and batch_size is not None:
             raise ValueError(
-                "`self.preprocessor` is `None`, please make sure "
-                "`preprocessor` is set before calling `generate`."
+                "When `inputs` is a `tf.data.Dataset`, please do not provide "
+                "`batch_size` (the dataset is expected to be batched). "
+                f"Received: `batch_size={batch_size}` (of type "
+                f"{type(batch_size)})."
             )
 
-        # Tokenize.
-        prompt = tf.convert_to_tensor(prompt)
-        input_is_scalar = prompt.shape.rank == 0
-        prompt = prompt[tf.newaxis] if input_is_scalar else prompt
-        prompt = self.preprocessor.tokenizer(prompt)
+        if not isinstance(inputs, tf.data.Dataset):
+            if self.preprocessor is not None:
+                inputs = tf.convert_to_tensor(inputs)
+                input_is_scalar = inputs.shape.rank == 0
+                inputs = inputs[tf.newaxis] if input_is_scalar else inputs
+
+            if batch_size is not None:
+                inputs = tf.data.Dataset.from_tensor_slices(inputs)
+                inputs = inputs.batch(batch_size)
+            else:
+                # If `batch_size` was not provided, just wrap in a list to avoid
+                # the overhead of converting to a dataset.
+                inputs = [inputs]
+
+        def preprocess(x):
+            return self.preprocessor(
+                x,
+                sequence_length=max_length,
+                return_labels=False,
+                add_end_token=False,
```

Comment:
Let's add a comment here to explain users are expected to do end token padding in string space.